multicluster/controllers/multicluster/commonarea/acnp_resourceimport_controller.go

```diff
@@ -0,0 +1,193 @@
+		return ctrl.Result{}, r.reportStatusEvent(msg, ctx, resImp)
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *ResourceImportReconciler) handleResImpDeleteForClusterNetworkPolicy(ctx context.Context, resImp *multiclusterv1alpha1.ResourceImport) (ctrl.Result, error) {
+	acnpName := types.NamespacedName{
+		Namespace: "",
+		Name:      common.AntreaMCSPrefix + resImp.Spec.Name,
+	}
+	klog.InfoS("Deleting ACNP corresponding to ResourceImport",
+		"acnp", acnpName.String(), "resourceimport", klog.KObj(resImp))
+
+	acnp := &v1alpha1.ClusterNetworkPolicy{}
+	err := r.localClusterClient.Get(ctx, acnpName, acnp)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(2).InfoS("ACNP corresponding to ResourceImport has already been deleted",
+				"acnp", acnpName.String(), "resourceimport", klog.KObj(resImp))
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	if err = r.localClusterClient.Delete(ctx, acnp, &client.DeleteOptions{}); err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+func getMCAntreaClusterPolicy(resImp *multiclusterv1alpha1.ResourceImport) *v1alpha1.ClusterNetworkPolicy {
+	if resImp.Spec.ClusterNetworkPolicy == nil {
+		return nil
+	}
+	return &v1alpha1.ClusterNetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: common.AntreaMCSPrefix + resImp.Spec.Name,
+			Annotations: map[string]string{
+				common.AntreaMCACNPAnnotation: "true",
+			},
+		},
+		Spec: *resImp.Spec.ClusterNetworkPolicy,
+	}
+}
+
+func (r *ResourceImportReconciler) reportStatusEvent(errMsg string, ctx context.Context, resImp *multiclusterv1alpha1.ResourceImport) error {
+	if errMsg == "" {
+		return nil
+	}
+	statusEvent := &corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      randName(acnpImportStatusPrefix + r.localClusterID + "-"),
+			Namespace: resImp.Namespace,
+		},
+		Type:    corev1.EventTypeWarning,
+		Reason:  acnpImportFailed,
+		Message: errMsg,
+		InvolvedObject: corev1.ObjectReference{
+			APIVersion: resourceImportAPIVersion,
+			Kind:       resourceImportKind,
+			Name:       resImp.Name,
+			Namespace:  resImp.Namespace,
+			UID:        resImp.GetUID(),
+		},
+		FirstTimestamp:      metav1.Now(),
+		LastTimestamp:       metav1.Now(),
+		ReportingController: acnpEventReportingController,
+		ReportingInstance:   acnpEventReportingInstance,
+		Action:              "synced",
+	}
+	if err := r.remoteCommonArea.Create(ctx, statusEvent, &client.CreateOptions{}); err != nil {
+		klog.ErrorS(err, "Failed to create ACNP import event for ResourceImport", "resImp", klog.KObj(resImp))
+		return err
+	}
+	return nil
+}
+
+func randSeq(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		// #nosec G404: random number generator not used for security purposes
+		randIdx := rand.Intn(len(lettersAndDigits))
+		b[i] = lettersAndDigits[randIdx]
+	}
+	return string(b)
+}
+
+func randName(prefix string) string {
+	return prefix + randSeq(nameSuffixLength)
+}
```

Comment:
I noticed there are a few duplicate code in test framework, can we share them somewhere? maybe address it in a separate PR.

Reply:
Sure